### ISSUE TITLE
SampleGen: iterate over `map` fields in response

### ISSUE THREAD

**Issue by beccasaurus** [googleapis/gapic-generator#2584]
Samples cannot currently access `map` fields in responses.

⚠️ `java.lang.IllegalArgumentException: AnalyzeEntities:language_entities_text: entity.metadata is not a repeated field`

![screen shot 2019-02-21 at 1 06 13 pm](https://user-images.githubusercontent.com/280/53201705-80683880-35d9-11e9-9400-990bbf6df7e2.png)

```yaml
- id: language_entities_text
  description: "Analyze entities in text"
  parameters:
    defaults:
    - document.type=PLAIN_TEXT
    - document.content="California is a state."
  on_success:
  - loop:
      collection: $resp.entities
      variable: entity
      body:
      #
      # 🤔 How can we access metadata?
      #
      - loop:
          collection: entity.metadata
          variable: item
          body:
          - print:
            - "What is item? %s"
            - item
```

**Comment by beccasaurus** [comment]
=> @hzyi-google 

/cc @tswast @vchudnov-g

**Comment by vchudnov-g** [comment]
Thanks for the report. We had decided to punt on maps until there was a use case for it. Now we'll tackle it!

**Comment by beccasaurus** [comment]
Here's an example GAPIC config you can use to test:
https://github.com/beccasaurus/gapic-docs-samples/blob/master/google/cloud/language/v1/language_gapic-with-iterating-over-map.yaml#L138-L149

**Comment by hzyi-google** [comment]
Fixed by #2593
